### ISSUE TITLE
Translation management for configurable fields

### ISSUE THREAD

**Issue by jfayad** [Geeks-Solutions/nuxt-sections#109]
The translation management should apply to the options of configurable sections.

Including Wysiwyg field types

**Comment by akramsarkis** [comment]
@jfayad Can you please give more details here,  do you mean that for configurable section when I switch from a language to another in a form, I must be able to enter different values for the same field, except for the drop down lists or numbers ?

**Comment by mahfouzhseineweev** [comment]
@jfayad This is going to a breaking change for the existing configurable sections which doesn't have translations.

The new translation management will try expect an object with translation keys and instead it will find the field values stored as strings.

So for this one I think we also need to support backward compatibility correct ?

**Comment by jfayad** [comment]
> So for this one I think we also need to support backward compatibility correct ?

Yes, but make it a configuration so that out of the box it should enforce the new structure

**Comment by mahfouzhseineweev** [comment]
So for this one we will also use the default language that we will introduce for the project

**Comment by akramsarkis** [comment]
### Acceptance testing

- [ ] For field in configurable section that are text or wyziwig; if there are multiple language, we should be able to enter translation for them
- [ ] In case the site has only one language that is the default language of the site, in the section i must see only the translation of this language
- [ ] In view mode if there is multiple language then when switching language the translations must show else only the one of the default language will show

**Comment by akramsarkis** [comment]
@mahfouzhseineweev don't forget to add automated test for this one and to tell which one they are

**Comment by akramsarkis** [comment]
@mahfouzhseineweev please find below my feedback

- [ ] For the Link CTA, when you click to add a section let say the selective article, it is showing inside [object Object]

![Image](https://github.com/user-attachments/assets/18021d2d-4980-4fb9-a8fb-ee2e0c2e04b5)

- [ ] Still for the selective article, for the cta link I am not able to have a link for english and one for french

**Comment by akramsarkis** [comment]
@mahfouzhseineweev I still have bugs

1) Product facets : Even though the field is filled in, it is returning the error message ( whatever language configuration you have, french alone, english alone, both available

2) Even though I filled in all the fields for Selective articles, when only french is selected for the project, there is the error that the fields are required

**Comment by mahfouzhseineweev** [comment]
@akramsarkis For the first point, I opened the form, I submitted and I got the error showing correctly.
Then I added an array value and resubmitted, the form was submitted correctly

**Comment by mahfouzhseineweev** [comment]
@akramsarkis 

2. Even though I filled in all the fields for Selective articles, when only french is selected for the project, and I add the section, there is the error that the fields are required

I you want to do that, you have to select your default language as `FR` and for now you can't do that because maiori is on sections BO prod which doesn't have this update yet. So you should try with Only `EN` for example and redo the test

**Comment by akramsarkis** [comment]
> [@akramsarkis](https://github.com/akramsarkis)
> 
> 2. Even though I filled in all the fields for Selective articles, when only french is selected for the project, and I add the section, there is the error that the fields are required
> 
> I you want to do that, you have to select your default language as `FR` and for now you can't do that because maiori is on sections BO prod which doesn't have this update yet. So you should try with Only `EN` for example and redo the test

Ok for this one , for english there is no problem I tested

**Comment by akramsarkis** [comment]
> [@akramsarkis](https://github.com/akramsarkis) For the first point, I opened the form, I submitted and I got the error showing correctly. Then I added an array value and resubmitted, the form was submitted correctly

Well this is weird, I tried several time and get the error without being able to submit, let me recheck

**Comment by akramsarkis** [comment]
@mahfouzhseineweev julien added on his local for selective articles the 2 wyziwyg  title and description but I can't add a translation for each, I have the same value

**Comment by akramsarkis** [comment]
all good now, this issue can be closed
